File: backend/src/main/java/com/yagubogu/stadium/repository/StadiumRepository.java
package com.yagubogu.stadium.repository;

import com.yagubogu.stadium.domain.Stadium;
import com.yagubogu.stadium.domain.StadiumLevel;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface StadiumRepository extends JpaRepository<Stadium, Long> {

    Optional<Stadium> findByShortName(String shortName);

    Optional<Stadium> findByLocation(String location);

    List<Stadium> findAllByLevel(StadiumLevel level);
}

File: android/app/src/main/java/com/yagubogu/domain/repository/StadiumRepository.kt
package com.yagubogu.domain.repository

import com.yagubogu.data.dto.response.stadium.StadiumsWithGamesResponse
import java.time.LocalDate

interface StadiumRepository {
    suspend fun getStadiumsWithGames(date: LocalDate): Result<StadiumsWithGamesResponse>
}
